skymap_scanner/client/reco_icetray.py

```diff
@@ -83,35 +83,33 @@ def save_to_disk_cache(frame: icetray.I3Frame, save_dir: Path) -> Path:
     return pixel_fname
 
 
-def get_GCD_diff_base_handle(baseline_GCD_file: str) -> Any:
+def get_baseline_GCD_handle(baseline_GCD_file: str) -> Any:
     """Find an available GCD base path."""
     stagers = dataio.get_stagers()
 
-    # try to load the base file from the various possible input directories
-    GCD_diff_base_handle = None
+    LOGGER.debug(f"Baseline GCD at {baseline_GCD_file}")
+
+    # assuming baseline_GCD_file is a valid GCD path!
+    # NOTE: old code use to search for a basename in a list of GCD_BASE_DIRS
+    baseline_GCD_handle = None
     if baseline_GCD_file not in [None, "None"]:
-        for GCD_base_dir in cfg.GCD_BASE_DIRS:
-            try:
-                read_url = os.path.join(GCD_base_dir, baseline_GCD_file)
-                LOGGER.debug("reading baseline GCD from {0}".format(read_url))
-                GCD_diff_base_handle = stagers.GetReadablePath(read_url)
-                if not os.path.isfile(str(GCD_diff_base_handle)):
-                    raise RuntimeError("file does not exist (or is not a file)")
-            except:
-                LOGGER.debug(" -> failed")
-                GCD_diff_base_handle = None
-            if GCD_diff_base_handle is not None:
-                LOGGER.debug(" -> success")
-                break
-
-        if GCD_diff_base_handle is None:
-            raise RuntimeError(
-                "Could not read the input GCD file '{0}' from any pre-configured location".format(
-                    baseline_GCD_file
-                )
-            )
+        try:
+            LOGGER.debug("reading baseline GCD from {0}".format(baseline_GCD_file))
+            baseline_GCD_handle = stagers.GetReadablePath(baseline_GCD_file)
+            if not os.path.isfile(str(baseline_GCD_handle)):
+                raise RuntimeError("file does not exist (or is not a file)")
+        except:
+            LOGGER.debug(" -> failed")
+            baseline_GCD_handle = None
+        if baseline_GCD_handle is not None:
+            LOGGER.debug(" -> success")
+
+    if baseline_GCD_handle is None:
+        raise RuntimeError(f"Could not read the input GCD file '{baseline_GCD_file}'")
+    else:
+        LOGGER.info("baseline_GCD_file is None. Likely frame packet is not a GCD diff.")
```

Comment:
I guess I don't follow the logic that `baseline_GCD_handle is None` implies `baseline_GCD_file is None`, at least without intense reading.